### DISCUSSION TITLE
git submodule add https://github.com/Lruihao/FixIt.git (NOT A GIT REPOSITORY)

### DISCUSSION THREAD

**Discussion by darrenaddy** [hugo-fixit/FixIt#110] [Q&A] [answered]
### Describe the bug 描述你遇到的错误

Attempting to follow the documentation (both in the README.md and the docs site). It says (to install as a submodule):
`git submodule add https://github.com/Lruihao/FixIt.git themes/FixIt`

returns:
`fatal: not a git repository (or any of the parent directories): .git`

### Build Environment 构建环境

Windows 10
Hugo v0.96.0 extended
Theme version: whatever version is current on 4/11/22 13:35 UTC

### Additional Information 补充信息

Thank you for developing this theme!
I love how well documented it is. I did not try to download the zip file and install that way, because I prefer to use the easily updatable git method.

**Comment by darrenaddy** [comment]
Please add the step to run:
`git init`
in the project folder **before** the step to run:
`git submodule add https://github.com/Lruihao/FixIt.git themes/Fixit`
for those of us with little previous git experience! Thank you.
Issue is solved, but I will leave open so that the request to add the step in the README.md and installation documentation is seen by the developer(s).

**Comment by Lruihao** [comment, marked as answer]
As your error message echo that `fatal: not a git repository (or any of the parent directories): .git`  
in other words, your project is not a git repository. So, you should execute the following command at first in you project folder

```bash
git init
```

then,

```bash
git submodule add https://github.com/Lruihao/FixIt.git themes/FixIt
```

**Comment by Lruihao** [comment]
> Please add the step to run: `git init` in the project folder **before** the step to run: `git submodule add https://github.com/Lruihao/FixIt.git themes/Fixit` for those of us with little previous git experience! Thank you. Issue is solved, but I will leave open so that the request to add the step in the README.md and installation documentation is seen by the developer(s).

yeap! but i think it's better to convert to a disscussion.

**Comment by darrenaddy** [comment]
Perhaps also consider offering the option to avoid git submodules and install as a Hugo Module:
[Hugo Modules for Themes - Instead of git submodules](https://broersma.dev/hugo-modules-for-themes/)

**Reply by Lruihao** [reply]
Yes, it depends on whether the user is more familiar with git or golang.

**Comment by darrenaddy** [comment]
Note: I also ran into this message when attempting to run Hugo server:
`Error: module "FixIt" not found; either add it as a Hugo Module or store it in "D:\\Hugo".: module does not exist`

I solved this by changing the themesDir in config.toml
`- themesDir = "../.."`
`+ themesDir = "themes"`

**Reply by Lruihao** [reply]
yea, the value of themesDir shall be subject to the theme module actual storage location.  
and the default value of themesDir is "themes", so just delete this line is okay.

```diff
- themesDir = "../.."
```

my [hugo-blog](https://github.com/Lruihao/hugo-blog) site maybe is a good example for you.

**Comment by darrenaddy** [comment]
Just providing feedback on the installation process.
Note the tip in documentation 2.5 says:

>When you run `hugo serve`, when the contents of the files change, the page automatically refreshes with the changes.

I believe this should say `hugo server`


When running the hugo server, I initially ran:
`hugo server -D --verbose`
(which I generally do). Hugo responded as it normally does (working fine).
Then I saw the instruction to:

> Since the theme use .Scratch in Hugo to implement some features, it is highly recommended that you add --disableFastRender parameter to hugo server command for the live preview of the page you are editing.

When I added --disableFastRender to my hugo server then I get this (before the list of pages, files & processed images):
`WARN 2022/04/11 09:11:56`
`Current environment is "development". The "comment system", "CDN" and "fingerprint" will be disabled.
当前运行环境是 "development". "评论系统", "CDN" 和 "fingerprint" 不会启用.`

I'm not sure if this is a problem, but I did want to use the CDN option that this theme provides.

**Reply by Lruihao** [reply]
> I believe this should say `hugo server`

oh no, it's a typo error (i will revise tomorrow)

---

> When I added --disableFastRender to my hugo server then I get this (before the list of pages, files & processed images):
>> WARN 2022/04/11 09:11:56
>> Current environment is "development". The "comment system", "CDN" and "fingerprint" will be disabled. 当前运行环境是 "development". "评论系统", "CDN" 和 "fingerprint" 不会启用.

this is not a problem, if you want use cdn in local environment, you should run `hugo server --environment production` or `hugo server -e production`

**Reply by darrenaddy** [reply]
Thank you! 谢谢

However when I run:
```
hugo server -e production
port 1313 already in use, attempting to use an available port
Start building sites …
hugo v0.96.0-2fd4a7d3d6845e75f8b8ae3a2a7bd91438967bbb+extended windows/amd64 BuildDate=2022-03-26T09:15:58Z VendorInfo=gohugoio
ERROR 2022/04/11 09:35:39 Failed to read Git log: fatal: your current branch 'master' does not have any commits yet
Error: Error building site: logged 1 error(s)
```

Googling, it appears that this is again because of my not being familiar with the ins-and-outs of git. Looks like I have to work through this (ridiculous amount of) git documentation:
[https://medium.com/@josephinechika94/getting-started-with-git-a4011f071105](https://medium.com/@josephinechika94/getting-started-with-git-a4011f071105)

**Reply by Lruihao** [reply]
> Failed to read Git log: fatal: your current branch 'master' does not have any commits yet

FixIt need .git info for getting posts modified date, but your project is a new git repo that your current branch 'master' does not have any commits yet.
so, you just to do that push a commit to master branch.

**Reply by Lruihao** [reply]
well... actually, `hugo serve` is an alias of `hugo server`,so not a typo error :D
![image](https://user-images.githubusercontent.com/33419593/162766041-3177dfef-1da4-4eea-9f2c-31c1b14c5dbb.png)

**Comment by darrenaddy** [comment]
One other (very minor) thing.
When I started hugo server and looked at the page in my browser, it looked correct except for a missing "images/avatar.png".
I thought that perhaps this was in the static/images/ directory of the exampleSite so I copied all exampleSite files over to the root folder.
"images/avatar.png" was still missing.

So I searched and found that avatar.png is found in the /assets/ directory. So I am guessing that this image will **not** be missing once we build the static site?

**Reply by Lruihao** [reply]
- in `assets/`, resource will not be missing once we used it.
- in `static/`, resource will always be there, whether we use it or not.
